src/Qir/Execution/Tools/QirTools.cs

```diff
@@ -19,9 +24,22 @@ public static class QirTools
         /// <param name="qsharpDll">.NET DLL generated by the Q# compiler.</param>
         /// <param name="libraryDirectory">Directory where the libraries to link to are located.</param>
         /// <param name="includeDirectory">Directory where the headers needed for compilation are located.</param>
-        public static Task BuildFromQSharpDll(FileInfo qsharpDll, DirectoryInfo libraryDirectory, DirectoryInfo includeDirectory)
+        public static async Task BuildFromQSharpDll(FileInfo qsharpDll, DirectoryInfo libraryDirectory, DirectoryInfo includeDirectory)
```

Comment:
Thinking about the signature of this method more closely, there's no way for the caller to know where the executables were placed. Maybe the following signature would be better:
```csharp
public static async Task<IList<FileInfo>> BuildFromQSharpDll(FileInfo qsharpDll, DirectoryInfo libraryDirectory, DirectoryInfo includeDirectory, DirectoryInfo executablesDirectory);
```
Where: 
- The method returns a list of the executables created.
- It receives a directory to place the executables as argument.

Reply:
I've implemented the executable directory parameter. From a conversation with you offline, it sounds like we want to wait on returning a list of FileInfos for the created executables.

Reply:
The part to return the list of executables will be done once the `BuildAsync` method returns the `FileInfo` of the executable.